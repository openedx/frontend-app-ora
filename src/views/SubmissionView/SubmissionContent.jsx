import React from 'react';

import { useORAConfigData } from 'data/services/lms/hooks/selectors';

import Prompt from 'components/Prompt';
import TextResponse from 'components/TextResponse';
import FileUpload from 'components/FileUpload';

const SubmissionContent = () => {
  const { prompts } = useORAConfigData();
  return (
    <div>
      {
        prompts.map((prompt, index) => (
          <div>
            <Prompt promptIndex={index} />
            <TextResponse promptIndex={index} />
          </div>
        ))
      }
      <FileUpload />
    </div>
  );
};

export default SubmissionContent;
