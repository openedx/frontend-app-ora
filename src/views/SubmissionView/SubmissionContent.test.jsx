import { shallow } from '@edx/react-unit-test-utils';
import SubmissionContent from './SubmissionContent';

jest.mock('@edx/paragon/icons', () => ({
  CheckCircle: 'CheckCircle',
}));

jest.mock('components/Prompt', () => 'Prompt');
jest.mock('components/TextResponse', () => 'TextResponse');
jest.mock('components/FileUpload', () => 'FileUpload');

describe('<SubmissionContent />', () => {
  const props = {
    submission: {
      response: {
        textResponses: ['test'],
        uploadedFiles: [{
          fileDescription: 'test',
        }],
      },
    },
    oraConfigData: {
      prompts: ['<p>test</p>'],
      submissionConfig: {
        maxFileSize: 100,
      },
    },
    onTextResponseChange: jest.fn().mockName('onTextResponseChange'),
    onFileUploaded: jest.fn().mockName('onFileUploaded'),
    draftSaved: true,
  };

  describe('render', () => {
    test('default', () => {
      const wrapper = shallow(<SubmissionContent {...props} />);
      expect(wrapper.snapshot).toMatchSnapshot();

      expect(wrapper.instance.findByType('Prompt')).toHaveLength(1);
    });

    test('no prompts', () => {
      const wrapper = shallow(<SubmissionContent {...props} oraConfigData={{ prompts: [] }} />);
      expect(wrapper.snapshot).toMatchSnapshot();

      expect(wrapper.instance.findByType('Prompt')).toHaveLength(0);
    });
  });
});
