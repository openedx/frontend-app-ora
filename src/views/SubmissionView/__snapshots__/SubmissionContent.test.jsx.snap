// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SubmissionContent /> render default 1`] = `
<div>
  <div
    className="d-flex justify-content-between"
  >
    <h2
      className="mb-4"
    >
      Your response
    </h2>
    <p
      className="d-flex text-primary"
    >
      <Icon
        src="CheckCircle"
      />
      Draft saved
    </p>
  </div>
  <p>
    <strong>
      Instructions
      : 
    </strong>
    Create a response to the prompt below.
    Progress will be saved automatically and you can return to complete your
    progress at any time. After you submit your response, you cannot edit
    it.
  </p>
  <div
    key="0"
  >
    <Prompt
      prompt="<p>test</p>"
    />
    <TextResponse
      onChange={[Function]}
      submissionConfig={
        Object {
          "maxFileSize": 100,
        }
      }
      value="test"
    />
  </div>
  <FileUpload
    onFileUploaded={[MockFunction onFileUploaded]}
    uploadedFiles={
      Array [
        Object {
          "fileDescription": "test",
        },
      ]
    }
  />
</div>
`;

exports[`<SubmissionContent /> render no prompts 1`] = `
<div>
  <div
    className="d-flex justify-content-between"
  >
    <h2
      className="mb-4"
    >
      Your response
    </h2>
    <p
      className="d-flex text-primary"
    >
      <Icon
        src="CheckCircle"
      />
      Draft saved
    </p>
  </div>
  <p>
    <strong>
      Instructions
      : 
    </strong>
    Create a response to the prompt below.
    Progress will be saved automatically and you can return to complete your
    progress at any time. After you submit your response, you cannot edit
    it.
  </p>
  <FileUpload
    onFileUploaded={[MockFunction onFileUploaded]}
    uploadedFiles={
      Array [
        Object {
          "fileDescription": "test",
        },
      ]
    }
  />
</div>
`;
